### ISSUE TITLE
Use a single widget wrapper in the Workflow view

### ISSUE THREAD

**Issue by kinow** [cylc/cylc-ui#461]
**Describe exactly what you would like to see in an upcoming release**

In the Workflow component (not view) every widget-ized component gets its own wrapper. So there is the Tree component, and the TreeWrapper. The Graph component, and the GraphWrapper.

Instead we should be able to use a single wrapper. 

**Additional context**

There are multiple ways to achieve this, I think.

We can wait until we have a uniform interface for components (i.e. Tree, Graph, Dot, etc, would all have the same props).

Or, we can use a default slot and allow each component to request whatever parameters are necessary.

I used the second approach when creating the Lumino Vue example app linked in the JuptyerLab project page: https://github.com/kinow/vue-lumino/blob/6f01f9b32717d267f1372838d1e0d2a73fa91ac4/src/components/Lumino.vue#L84

I think that `VueComponentWrapper` can be ported back to Cylc UI, fixing this issue.

**Pull requests welcome!**

**Comment by kinow** [comment]
Managed to write a version of the component that doesn't need any wrapper at all. The approach to sync the DOM with Vue's virtual DOM is a bit different.

Before we had a wrapper component that receives the props/events/etc, and then uses its lifecycle hooks to control when the DOM element was created/destroyed/etc.

Now, we use the default slot `<slot></slot>` in the `Lumino` component. But it will be under a `<div v-show=false>`, so they are never displayed. Instead we display the Lumino DOM widgets, which start empty.

Then for each child component of the `Lumino` component, it uses its `updated` lifecycle function to check which children it needs to add/update, and then moves the Vue `$el` (HTML DOM element of a Vue component) to the Lumino widget `DockPanel`.

I'm pushing this component to NPM, but won't use it in Cylc. Instead will duplicate the code (some 100's lines of code only) and customize it very little (e.g. the tab title is a bit different in Cylc, while I'm using the Component `.name` property, but not super important :+1: )

TL;DR: we should be able to delete a bunch of code that is probably one of the most confusing to maintain in the UI (assuming everything works well)
